apps/services/user-profile/src/app/worker/test/worker.spec.ts

```diff
@@ -0,0 +1,569 @@
+
+import { UserProfile } from '../../user-profile/userProfile.model'
+
+import { ProcessedStatus } from '../types'
+
+describe('UserProfileWorker', () => {
+  jest.setTimeout(30000)
+  let app: TestApp
+  let workerService: UserProfileWorkerService
+  let userProfileModel: typeof UserProfile
+  let userProfileAdvaniaModel: typeof UserProfileAdvania
+
+  beforeEach(async () => {
+    app = await testServer({
+      appModule: UserProfileWorkerModule,
+      hooks: [
+        useDatabase({ type: 'postgres', provider: SequelizeConfigService }),
+      ],
+    })
+    workerService = app.get(UserProfileWorkerService)
+    userProfileModel = app.get(getModelToken(UserProfile))
+    userProfileAdvaniaModel = app.get(getModelToken(UserProfileAdvania))
+  })
+
+  it('should import new profiles', async () => {
+    // Arrange
+    const advaniaProfiles = new Array(10).fill(null).map((_, index) => ({
+      ssn: `${index}`,
+      email: `test${index}@test.local`,
+      mobilePhoneNumber: `888111${index}`,
+      exported: new Date(2023, 10, 1),
+    }))
+
+    // Act
+    await userProfileAdvaniaModel.bulkCreate(advaniaProfiles)
+    const advaniaProfilesBeforeMigration =
+      await userProfileAdvaniaModel.findAll()
+    const userProfilesBeforeMigration = await userProfileModel.findAll()
+    await workerService.run()
+    const advaniaProfilesAfterMigration =
+      await userProfileAdvaniaModel.findAll()
+    const userProfilesAfterMigration = await userProfileModel.findAll()
+
+    // Assert
+    expect(
+      advaniaProfilesBeforeMigration.every(
+        (p) => p.status === ProcessedStatus.PENDING,
+      ),
+    ).toBe(true)
+    expect(userProfilesBeforeMigration.length).toBe(0)
+    expect(
+      advaniaProfilesAfterMigration.every(
+        (p) => p.status === ProcessedStatus.DONE,
+      ),
+    ).toBe(true)
+    expect(
+      userProfilesAfterMigration.every((p) => {
+        const matchingProfile = advaniaProfilesAfterMigration.find(
+          (ap) => ap.ssn === p.nationalId,
+        )
+
+        return (
+          matchingProfile.email === p.email &&
+          matchingProfile.mobilePhoneNumber === p.mobilePhoneNumber
+        )
+      }),
+    ).toBe(true)
+  })
+
+  describe('profiles being migrated exist in user_profile', () => {
+    it('should keep profiles as is if email and phone numbers match but set nudge to nudgeLastAsked', async () => {
+      // Arrange
+      const advaniaProfiles = new Array(10).fill(null).map((_, index) => ({
+        ssn: `${index}`,
+        email: `test${index}@test.local`,
+        mobilePhoneNumber: `888111${index}`,
+        exported: new Date(2023, 10, 1),
+        nudgeLastAsked: new Date(2023, 10, 5),
+      }))
+
+      const existingProfiles = advaniaProfiles.map((p) => ({
+        nationalId: p.ssn,
+        email: p.email,
+        mobilePhoneNumber: p.mobilePhoneNumber,
+        modified: new Date(2023, 0, 1),
+        lastNudge: new Date(), // not null
```

Comment:
~ Existing profiles in our DB would have lastNudge as null as v1 is not storing it. But pretty sure this doesn't matter for this test case.